apiml-security-common/src/main/java/org/zowe/apiml/security/common/content/AbstractSecureContentFilter.java

```diff
@@ -79,6 +79,14 @@ protected void doFilterInternal(@NonNull HttpServletRequest request, @NonNull Ht
                 failureHandler.onAuthenticationFailure(request, response, authenticationException);
             } catch (RuntimeException e) {
                 resourceAccessExceptionHandler.handleException(request, response, e);
+            } finally {
+                Authentication authentication = authenticationToken.get();
```

Comment:
Why do you obtain the token from the holder instead of using the variable `authentication`. Keep in mind that this value could be changed during the processing for any reason.